packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/pipeline-parser.spec.ts

```diff
@@ -0,0 +1,152 @@
+import { expect } from 'chai';
+import PipelineParser from './pipeline-parser';
+
+const pipelines = [
+  {
+    usecase: 'no stage disabled',
+    input: `[{$unwind: "users"}]`,
+    output: [
+      '[',
+      '  {',
+      '    $unwind: "users",',
+      '  },',
+      ']'
+    ].join('\n')
+  },
+  {
+    usecase: 'all stages disabled',
+    input: `[\n//{$unwind: "users"},\n // {$limit: 20}\n]`,
+    output: [
+      '[',
+      '  // {',
+      '  //   $unwind: "users",',
+      '  // }',
+      '  // {',
+      '  //   $limit: 20,',
+      '  // }',
+      ']'
+    ].join('\n')
+  },
+  {
+    usecase: 'enabled first and last stage',
+    input: `[{$unwind: "users"},\n // {$limit: 20},\n {$sort: {name: -1}}]`,
+    output: [
+      '[',
+      '  {',
+      '    $unwind: "users",',
+      '  }, // {',
+      '  //   $limit: 20,',
+      '  // }',
```

Comment:
Did you have a chance to try out a fix we discussed? I did a small experiment today and seems like a combination of `retainLines` option + setting the start line even for the first comment seems to work:

<img src="https://user-images.githubusercontent.com/5036933/195280876-a8c68047-f55f-4bb3-a842-6cc61c66390b.png" width=480 />

<img src="https://user-images.githubusercontent.com/5036933/195280957-e4e665ea-f9c3-4df9-a92f-131c8d98557c.png" width=120 />

Reply:
Linking follow up tickets: 

1. [COMPASS-6191](https://jira.mongodb.org/browse/COMPASS-6191) PipelineParser: preserve block comments when converting back and forth between editors
2. [COMPASS-6192](https://jira.mongodb.org/browse/COMPASS-6192) PipelineParser: leading comments of a stage should always start on a new line